source/server/hot_restart_nop_impl.h

```diff
@@ -14,7 +17,14 @@ namespace Server {
  */
 class HotRestartNopImpl : public Server::HotRestart {
 public:
-  HotRestartNopImpl() {}
+  HotRestartNopImpl() {
+    int num_stats = 100;
+    const SharedMemoryHashSetOptions options = sharedMemHashOptions(num_stats);
+    ptr_ = static_cast<uint8_t*>(
+        calloc(SharedMemoryHashSet<Stats::RawStatData>::numBytes(options), num_stats));
+    stats_allocator_ = std::make_unique<Stats::BlockRawStatDataAllocator>(options, true, ptr_);
+  }
+  ~HotRestartNopImpl() { ::free(ptr_); }
```

Comment:
Be careful, here.  I'd be sure that that `stats_allocator_` doesn't attempt to write to this memory as it's destructing.  This is an artifact of the C++ order of destruction where the destructor runs first, and then the member variables are destroyed.  If this is a problem, we can modify the destructor to look like this:
```
~HotRestartNopImpl() { 
stats_allocator_ = nullptr;
::free(ptr_); 
}

Reply:
I'll do this -- is this the kind of bug I can write a test for? Or is this just intuition that this could fail?